src/app/pages/add-course/add-course.page.ts

```diff
@@ -1,33 +1,62 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import {
   FormBuilder,
   FormControl,
   FormGroup,
   Validators,
 } from '@angular/forms';
-import { Router } from '@angular/router';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SiteMap } from '@enums/site-map.enum';
+import { Course } from '@interfaces/course';
 import { CoursesService } from '@services/courses/courses.service';
+import { format } from 'date-fns';
+import { take } from 'rxjs/operators';
 
 @Component({
   selector: 'app-add-course',
   templateUrl: './add-course.page.html',
   styleUrls: ['./add-course.page.scss'],
 })
-export class AddCourseComponent {
-  addCourseForm: FormGroup = this.createForm();
+export class AddCourseComponent implements OnInit {
+  addCourseForm: FormGroup;
+  courseId: string;
 
   constructor(
     private readonly fb: FormBuilder,
     private readonly router: Router,
     private readonly coursesService: CoursesService,
+    private readonly route: ActivatedRoute,
   ) {}
 
-  createForm(): FormGroup {
+  ngOnInit(): void {
+    this.courseId = this.route.snapshot.paramMap.get('id');
+    if (!!this.courseId) {
+      this.coursesService
+        .getItemById(this.courseId)
+        .pipe(take(1))
```

Comment:
probably should not be needed - this will be http call in the end or something, so it will just return one result and complete. Sometimes you have to make assumptions about the observable under the hood. E.g. HttpClient observables are cold, and after returning results they complete - they work like a promise